nirc_ehr/test/src/org.labkey.test/tests.nirc_ehr/NIRC_EHRTest.java

```diff
@@ -364,54 +400,86 @@ private void createUser(String userEmail, String groupName, @Nullable String per
+
+        goToModule("Dumbster");
+        EmailRecordTable notifications = new EmailRecordTable(this);
+        waitForTextWithRefresh(WAIT_FOR_PAGE + WAIT_FOR_JAVASCRIPT, "Death Notification: " + aliveAnimalId); //wait for more than a min
+        notifications.getMessage("Death Notification: " + aliveAnimalId).getBody().
+                contains("Animal '" + aliveAnimalId +"' has been declared dead on '" + LocalDateTime.now().format(_dateFormat) + "'.");
+        notifications.clickMessage(notifications.getMessageWithSubjectContaining("Death Notification: " + aliveAnimalId));
+        String url = Locator.linkWithText("Click here to record Necropsy").findElement(notifications).getAttribute("href");
+
+        log("Entering Necropsy");
+        impersonate(NIRC_BASIC_SUBMITTER_VET);
+        beginAt(url);
+        waitForElement(Locator.name("necropsyWeight"));
+        setFormElement(Locator.name("necropsyWeight"), "23");
+
+        log("Entering Tissue Disposition");
+        Ext4GridRef tissueDisposition = _helper.getExt4GridForFormSection("Tissue Disposition");
+        _helper.addRecordToGrid(tissueDisposition);
+        tissueDisposition.setGridCell(1, "necropsyDispositionCode", "Frozen");
+        tissueDisposition.setGridCell(1, "necropsyTissue", "Pancreas");
+        waitAndClick(_helper.getDataEntryButton("Submit for Review"));
+
+        log("Assigning the reviewer");
+        Window<?> submitForReview = new Window<>("Submit For Review", getDriver());
+        setFormElement(Locator.tagWithNameContaining("input", "ehr-usersandgroups"), NIRC_FULL_SUBMITTER_VET);
+        submitForReview.clickButton("Submit");
+        stopImpersonating();
+
+        log("Verify rows were inserted in appropriate datasets");
+        goToEHRFolder();
+        verifyRowCreated("study", "necropsy", aliveAnimalId, 1);
+        verifyRowCreated("study", "grossPathology", aliveAnimalId, 9);
+        verifyRowCreated("study", "tissueDisposition", aliveAnimalId, 1);
 
-        //verify data in study.weight is inserted for this dead animal (submit final sets the QC State to "Completed", aka makes data public to be then available to view from Demographics as Most Recent Weight)
+        goToEHRFolder();
+        impersonate(NIRC_FULL_SUBMITTER_VET);
+        EnterDataPage enterDataPage = EnterDataPage.beginAt(this, getContainerPath());
+        enterDataPage.clickAllTasksTab();
+        waitAndClick(Locator.linkWithText("Death/Necropsy"));
+        switchToWindow(1);
+        submitForm("Submit Final", "Finalize");
+        switchToMainWindow();
+        stopImpersonating();
+
+        log("Verify rows were inserted in appropriate datasets");
+        goToEHRFolder();
+        verifyRowCreated("study", "weight", aliveAnimalId, 1);
 
-        //verify that animal's record is closed, meaning 'end date' is set in study.assignments, study.protocolAssignments, and study.housing as the date of death
```

Comment:
I don't see this being verified. If not this time around, then add this comment as a TODO

Reply:
There are no records in study.assignments, study.housing and study.protocolAssignments

Should i insert them before marking animal dead?